geode-core/src/main/java/org/apache/geode/internal/cache/TXState.java

```diff
@@ -932,13 +932,19 @@ void doCleanup() {
       synchronized (this.completionGuard) {
         this.completionGuard.notifyAll();
       }
-
+      if (isBucketLocks()) {
```

Comment:
not knowing complete logic; just wondering, if this check should come before notifying the completionGuard (above check).